base/linalg/cholesky.jl

```diff
@@ -86,7 +86,9 @@ and return the UpperTriangular matrix `U` such that `A = U'U`.
 """
 function chol{T}(A::AbstractMatrix{T})
     S = promote_type(typeof(chol(one(T))), Float32)
-    chol!(copy_oftype(A, S))
+    AA = similar(A, S, size(A))
+    copy!(AA, A)
```

Comment:
I noticed that we use `copy_oftype` in places where we don't want the fully mutable matrix, e.g. https://github.com/JuliaLang/julia/blob/master/base/linalg/tridiag.jl#L128. I thought about `mutablecopy` but decided that it was almost as simple just to write it out.